names/names.py

```diff
@@ -10,11 +56,25 @@
 names_2 = f.read().split("\n")  # List containing 10000 names
 f.close()
 
+#mvp
 duplicates = []
-for name_1 in names_1:
-    for name_2 in names_2:
-        if name_1 == name_2:
-            duplicates.append(name_1)
+
+BinaryName = BinarySearchTree('new_name')
+for i in names_1:
+    BinaryName.insert(i)
+
+for j in names_2:
+    if BinaryName.contains(j):
+        duplicates.append(j)
+
```

Comment:
You'll be using and learning about sets a lot this week. Think about how you might refactor lines 61-70 using sets.